File: problemas/1000/1094.php
<?php

$n = readline();

$cobaias = [
    'C' => 0,
    'R' => 0,
    'S' => 0
];

for ($i = 0; $i < $n; $i++) {
    $linha = explode(' ', readline());
    $cobaias[$linha[1]] += $linha[0];
}

$total = array_sum($cobaias);
$divisor = $total / 100;

echo 'Total: ' . $total . " cobaias\n";
echo 'Total de coelhos: ' . $cobaias['C'] . "\n";
echo 'Total de ratos: ' . $cobaias['R'] . "\n";
echo 'Total de sapos: ' . $cobaias['S'] . "\n";
echo 'Percentual de coelhos: ' . number_format($cobaias['C'] / $divisor, 2, '.', '' ) . " %\n";
echo 'Percentual de ratos: ' . number_format($cobaias['R'] / $divisor, 2, '.', '' ) . " %\n";
echo 'Percentual de sapos: ' . number_format($cobaias['S'] / $divisor, 2, '.', '' ) . " %\n";

File: problemas/1000/1094.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace uri1094
{
    internal static class Program
    {
        private static void Main()
        {
            int n = int.Parse(Console.ReadLine());

            var cobaias = new Dictionary<char, int>
            {
                {'C', 0},
                {'R', 0},
                {'S', 0}
            };

            for (int i = 0; i < n; i++)
            {
                string[] linha = Console.ReadLine().Split(' ');
                cobaias[linha[1][0]] += int.Parse(linha[0]);
            }

            int total = cobaias.Sum(x => x.Value);
            double divisor = total / 100.0;

            Console.WriteLine($"Total: {total} cobaias");
            Console.WriteLine($"Total de coelhos: {cobaias['C']}");
            Console.WriteLine($"Total de ratos: {cobaias['R']}");
            Console.WriteLine($"Total de sapos: {cobaias['S']}");
            Console.WriteLine($"Percentual de coelhos: {cobaias['C'] / divisor:F2} %");
            Console.WriteLine($"Percentual de ratos: {cobaias['R'] / divisor:F2} %");
            Console.WriteLine($"Percentual de sapos: {cobaias['S'] / divisor:F2} %");
        }
    }
}